providers/azure_private_dns/azurePrivateDnsProvider.go

```diff
@@ -0,0 +1,654 @@
+					rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+					rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+					rc.Type = "NS"
+					_ = rc.SetTarget(*rec.Nsdname)
+					results = append(results, rc)
+				}
+		*/
+	case "Microsoft.Network/privateDnsZones/PTR":
+		for _, rec := range set.Properties.PtrRecords {
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "PTR"
+			_ = rc.SetTarget(*rec.Ptrdname)
+			results = append(results, rc)
+		}
+	case "Microsoft.Network/privateDnsZones/TXT":
+		if len(set.Properties.TxtRecords) == 0 { // Empty String Record Parsing
+			// This is a null TXT record.
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "TXT"
+			_ = rc.SetTargetTXT("")
+			results = append(results, rc)
+		} else {
+			// This is a normal TXT record. Collect all its segments.
+			for _, rec := range set.Properties.TxtRecords {
+				rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+				rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+				rc.Type = "TXT"
+				var txts []string
+				for _, txt := range rec.Value {
+					txts = append(txts, *txt)
+				}
+				_ = rc.SetTargetTXTs(txts)
+				results = append(results, rc)
+			}
+		}
+	case "Microsoft.Network/privateDnsZones/MX":
+		for _, rec := range set.Properties.MxRecords {
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "MX"
+			_ = rc.SetTargetMX(uint16(*rec.Preference), *rec.Exchange)
+			results = append(results, rc)
+		}
+	case "Microsoft.Network/privateDnsZones/SRV":
+		for _, rec := range set.Properties.SrvRecords {
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "SRV"
+			_ = rc.SetTargetSRV(uint16(*rec.Priority), uint16(*rec.Weight), uint16(*rec.Port), *rec.Target)
+			results = append(results, rc)
+		}
+		/* CAA doesn't apply to private zones
+		 	case "Microsoft.Network/dnszones/CAA":
+				for _, rec := range set.Properties.CaaRecords {
+					rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+					rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+					rc.Type = "CAA"
+					_ = rc.SetTargetCAA(uint8(*rec.Flags), *rec.Tag, *rec.Value)
+					results = append(results, rc)
+				}
+		*/
+	case "Microsoft.Network/privateDnsZones/SOA":
+	default:
+		panic(fmt.Errorf("nativeToRecords rtype %v unimplemented", *set.Type))
+	}
+	return results
+}
+
+// NOTE recordToNativeDiff2 is really "convert []RecordConfig to rrset".
+
+func (a *azurednsProvider) recordToNativeDiff2(recordKey models.RecordKey, recordConfig []*models.RecordConfig) (*adns.RecordSet, adns.RecordType, error) {
```

Comment:
Please do a global search and replace of `recordToNativeDiff2` to `recordToNative`